### PR TITLE
1160: use Eclipse-Job-API for repository actions

### PR DESCRIPTION
**Pull request by lostiniceland**
To give the user some visual feedback for long running
repository-actions (e.g. "Download All") the Runnable is wrapped with
the Eclipse-Job-API.

This will only give the user feedback that there is a job running and when it is done. 
In order to add more feedback with a progressbar the bnd-side must be enhanced with some other interface than Runnable (maybe some simple callback-interface which reveives an int for the current status).

Signed-off-by: Marc Schlegel <maschlegel@gmail.com>

### CODE REVIEW

**Comment by bjhargrave** [comment]
Please review @njbartlett.

File: bndtools.core/src/bndtools/views/repository/RepositoriesView.java [hunk 1]
```diff
@@ -613,17 +618,42 @@ public void menuAboutToShow(IMenuManager manager) {
...
+                                            Job backgroundJob = new Job("Repository Action '" + getText() + "'") {
+
+                                                @Override
+                                                protected IStatus run(IProgressMonitor monitor) {
+                                                    try {
+                                                        e.getValue().run();
+                                                        if (rp != null && rp instanceof Refreshable)
+                                                            Central.refreshPlugin((Refreshable) rp);
+                                                    } catch (final Exception e) {
+                                                        Display.getDefault().asyncExec(new Runnable() {
```
Comment (bjhargrave) [inline bndtools.core/src/bndtools/views/repository/RepositoriesView.java]: Should this really be a dialog? Or just put in the error log? I doubt there is much the user can do here except dismiss the dialog.

Comment (lostiniceland) [inline bndtools.core/src/bndtools/views/repository/RepositoriesView.java]: Right, the error-log is the better place to put this.

**Comment by bjhargrave** [comment]
Can you please squash into a single commit without a merge commit? If that exceeds your git fu, let me know and I can do it.

**Pull request closed without merging**